internal/services/postgres/postgresql_flexible_server_resource.go

```diff
@@ -278,7 +278,7 @@ func resourcePostgresqlFlexibleServer() *pluginsdk.Resource {
 
 			"public_network_access_enabled": {
 				Type:     pluginsdk.TypeBool,
-				Computed: true,
+				Optional: true,
 			},
```

Comment:
This is going to be a breaking change for existing users? as it will now set everyone to false whenthe default iirc is true?

as such we should make it optional + computed to account for existing resources + users changing the default outside terraform

Reply:
Yes. When user created the resource without `delegated_subnet_id`, `private_dns_zone_id` and `public_network_access_enabled` using old AzureRM version and then user upgraded to new AzureRM version, `tf plan` would cause difference if `public_network_access_enabled` isn't set in the tf config. Though there is TF difference but it's expected since service API would return the default value `true` for `public_network_access_enabled` when `delegated_subnet_id` and `private_dns_zone_id` aren't set.

OK. I will use optional + computed. Then the existing resources wouldn't be impacted anymore.